### ISSUE TITLE
Support "include" feature

### ISSUE THREAD

**Issue by takahi-i** [walter-cd/walter#120]
I would like Walter to support "include" feature which is a similar to Ansible include feature. This feature is useful to split large pipeline setting and in addition, reuse the settings.

I have written the design note for the feature. It would be very helpful if you (@walter-cd/owners)  gives any comments.

The following is the design note.

https://gist.github.com/takahi-i/ee34032c708e5de0f75e

**Comment by mizzy** [comment]
:+1:

**Comment by takahi-i** [comment]
Thank you very much for reviewing design note @mizzy!

**Comment by ainoya** [comment]
>  pipeline:
>  - include: tasks/git-clone.yml URL=git@github.com:recruit-tech/redpen.git ...(1)
>  - include: tasks/redpen-deploy VERSION=1.2                                             ...(2)

How can we write `pipeline.yml` when we want to execute (1), (2) in parallel ?

**Comment by takahi-i** [comment]
Thank you a lot for the comment @ainoya! Currently running multiple "include" files is not considered. If we want to run the included pipeline in parallel the format should be changed. 


One idea is that "include" is supported as a stage type. The following is the image of the pipeline configuration with parallel execution of "include" stages

```
pipeline:
   - name: deploy in parallel
     paralel:
         - name: build
           type: include
           file: tasks/build.yml
         - name: deploy
           type: include
           file: tasks/deploy.yml
```

Any comments would be appreciate it.

**Comment by ainoya** [comment]
I think this issue is related to #50, your solution is what split big pipeline workflow into small workflows.
But, it has another solution, is what "separate flow and statement" like below:

```
# s1_stages.yml
-------------
stage_definition:
  - name: s1
    command: echo 'hello stage1 with VAR1=${VAR1}'
    
# s2_stages.yml
-------------
stage_definition:
  - name: s2
    command: echo 'hello stage2'
  - name: s2-1
    command: echo 'hello stage2-1'
  - name: s2-2
    command: echo 'hello stage2-2'
   
# s3_stages.yml
-------------
stage_definition:
  - name: s3
    command: echo 'hello stage3'
 - name: s3-4
    command: echo 'hello stage3-4'

# pipeline.yml
-------------
include:
  - ./s1_stages.yml
  - ./s2_stages.yml
  - ./s3_stages.yml
    
pipeline:
  - name: s1
    vars:
      VAR1: aaaa
  - name: s2
    parallel:
      - name: s2-1
      - name: s2-2
  - name: s3
      - name: s3-4
```

Which do you prefer these solutions ?

**Comment by takahi-i** [comment]
Thank you very much for the the format idea @ainoya!

**Comment by takahi-i** [comment]
I think the @ainoya's is safer than mine. I added another format idea, which slightly change the @ainoya's format.  

https://gist.github.com/takahi-i/691c99c1dcb8ea7d7067

**Comment by takahi-i** [comment]
I uploaded another idea which support namespace. 

https://gist.github.com/takahi-i/66fc4dc5dcba5314996d

**Comment by takahi-i** [comment]
I will gives it a try with the third.

**Comment by ainoya** [comment]
:thumbsup:

**Comment by takahi-i** [comment]
I found a problem in the third design. I fixed the design 3 and try with the following design (design 4).

https://gist.github.com/takahi-i/f58b969216a4dec7e064

**Comment by takahi-i** [comment]
Fixed with #123
